rotkehlchen/tests/api/test_history_base_entry.py

```diff
@@ -43,11 +44,47 @@
     from rotkehlchen.api.server import APIServer
 
 
+def assert_editing_works(
+        entry: 'HistoryBaseEntry',
+        rotkehlchen_api_server: 'APIServer',
+        sequence_index: int,
+):
+    """Assert that editing any subclass of history base entry is successful done"""
+    entry.sequence_index = sequence_index
+    entry.timestamp = TimestampMS(1639924575000)
+    entry.balance = Balance(amount=FVal('1500.1'), usd_value=FVal('1499.45'))
+    entry.location_label = '0x9531C059098e3d194fF87FebB587aB07B30B1306'
+    entry.asset = A_USDT
+    entry.notes = 'Deposit stuff for staking somewhere'
+    if entry.entry_type in (HistoryBaseEntryType.EVM_EVENT, HistoryBaseEntryType.HISTORY_EVENT):
+        entry.event_type = HistoryEventType.DEPOSIT
+        entry.event_subtype = HistoryEventSubType.NONE
+    json_data = entry_to_input_dict(entry, include_identifier=True)
+    response = requests.patch(
+        api_url_for(rotkehlchen_api_server, 'historyeventresource'),
+        json=json_data,
+    )
+    assert_simple_ok_response(response)
+    assert entry.identifier is not None
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'historyeventresource',
+        ),
+        json={'event_identifiers': [entry.event_identifier]},
+    )
+    result = assert_proper_response_with_result(response)
+    for result_entry in result['entries']:
+        if result_entry['entry']['identifier'] == entry.identifier:
+            for valid_key in entry_keys[entry.entry_type]:
+                assert result_entry['entry'][valid_key] == entry.serialize()[valid_key]
```

Comment:
I believe to have mentioned it in other review but if you are not testing the api avoid making calls. It executes extra logic that you don't need. Here you can query the db directly using a filter and `get_history_events`. It will return the same information + you don't need to serialize the object if I'm correct